extensions/ql-vscode/src/vscode-tests/cli-integration/remote-queries/variant-analysis-manager.test.ts

```diff
@@ -156,13 +158,177 @@ describe('Variant Analysis Manager', async function() {
       it('should pop download tasks off the queue', async () => {
         const getResultsSpy = sandbox.spy(variantAnalysisManager, 'autoDownloadVariantAnalysisResult');
 
-        await variantAnalysisManager.enqueueDownload(scannedRepos[0], variantAnalysis, cancellationTokenSource.token);
-        await variantAnalysisManager.enqueueDownload(scannedRepos[1], variantAnalysis, cancellationTokenSource.token);
-        await variantAnalysisManager.enqueueDownload(scannedRepos[2], variantAnalysis, cancellationTokenSource.token);
+        await variantAnalysisManager.enqueueDownload(scannedRepos[0], variantAnalysisApiResponse, cancellationTokenSource.token);
+        await variantAnalysisManager.enqueueDownload(scannedRepos[1], variantAnalysisApiResponse, cancellationTokenSource.token);
+        await variantAnalysisManager.enqueueDownload(scannedRepos[2], variantAnalysisApiResponse, cancellationTokenSource.token);
 
         expect(variantAnalysisManager.downloadsQueueSize()).to.equal(0);
         expect(getResultsSpy).to.have.been.calledThrice;
       });
     });
   });
+
+  describe('when rehydrating a query', async () => {
+    let variantAnalysis: VariantAnalysis;
+    let variantAnalysisRemovedFired = false;
+    let onVariantAnalysisRemovedListener: Disposable;
+    let monitorVariantAnalysisCommandCalled = false;
+
+    beforeEach(() => {
+      variantAnalysis = createMockVariantAnalysis();
+
+      variantAnalysisRemovedFired = false;
+      onVariantAnalysisRemovedListener = variantAnalysisManager.onVariantAnalysisRemoved(() => {
+        variantAnalysisRemovedFired = true;
+      });
```

Comment:
If you use a `sinon.stub` instead of using an arrow function, you should be able to assert that the function was called using [`sinon.assert`](https://sinonjs.org/releases/latest/assertions/)

Reply:
Thanks. That's very handy to know.